docs/advanced/writing-declarations.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Writing Declarations
+---
+
+The real power of the transpiler is unlocked when combining it with declarations for your target environment. Declarations tell TypeScript which Lua API is available in your target context.
+
+If you need tips or help writing declarations, feel free to [join our Discord](https://discord.gg/BWAq58Y).
+
+## About Declaration Files
+
+Declaration files end with the extension `.d.ts`. These contain pure ambient code.
+
+For TypeScriptToLua, these files should contain information that describes the target Lua environment.
+
+This means functions, modules, variables and other members of the target Lua environment are primarily described in these files.
+
+They don't contain code that you would execute. Similar to how you'd write an interface in some other languages. TypeScriptToLua doesn't output any information from these files either.
+
+:::note
+You can write ambient declarations inside `.ts` files as well.
+:::
+
+## Declare Keyword
+
+The _declare_ keyword is used to say that the following declaration defines something that exists within global scope. Like something within the _\_G_ table in Lua.
+
+This is useful for defining Lua's environment.
+
+```ts title=_G.d.ts
+// Uses some declarations from
+// https://www.lua.org/manual/5.1/manual.html
+
+/**
+ * A global variable (not a function) that holds a string containing the
+ * current interpreter version.
+ */
+declare const _VERSION: number;
+
+/**
+ * Receives any number of arguments, and prints their values to stdout, using the
+ * tostring function to convert them to strings. print is not intended for
+ * formatted output, but only as a quick way to show a value, typically for
+ * debugging. For formatted output, use string.format.
+ * @param args Arguments to print
+ */
+declare function print(...args: any[]): void;
+```
+
+```ts title=main.ts
+print(_VERSION); // Editor and transpiler know what print and _VERSION are
+```
+
+:::note
+You can use _declare_ to write ambient declarations inside `.ts` files.
```

Comment:
Would make more sense to put `declare` (actual code) in monospacing, and _.ts_ italic.